.github/workflows/md-links.yml

```diff
@@ -0,0 +1,16 @@
+name: Markdown Link Checks
+on: pull_request
```

Comment:
I’d rather have this scheduled, instead of on each pull request. We discussed this in the automation call, but I’ll repeat my reasoning here for readers who weren’t there: pull requests shouldn’t be required to fix things outside of what they’re changing, and adding link checking for each pull request will result in pull requests having to fix links which happen to break (or wait for another PR fixing the broken link to be merged). Put another way, broken link checks don’t check for problems in the PR, unless they can limit their checks to links added or edited by the PR.

It would be great if this could be run periodically and open issues (using https://github.com/marketplace/actions/issue-from-template for example) when broken links are identified.

Reply:
Ah right, I got excited and unmarked as draft too soon (once the other issues were fixed).

Reply:
I understand the reasoning but we have the same paradigm with unit and e2e tests, ie it runs all tests for all PRs regardless of what was changed in the PR. In theory, a PR really shouldn't encounter an orthogonal  broken link as that should've been detected prior unless the link later becomes stale. But that shouldn't be common and the affected PR does not need to be held up by an orthogonal issue - we've done that many times for unrelated test failures. I think I prefer to detect a bad link before it gets merged.

Reply:
How about the current version, which does both? It only checks modified files on PRs, to try to limit the issues @skitt pointed out, but also would find newly-introduced errors as @tpantelis pointed out (I agree with both). I haven't added the Issue-raising logic yet, that might be better for a follow-up PR.

Reply:
The point of running all unit tests and e2e tests for all PRs is that it’s impossible to determine ahead of time, reliably, whether a code change is liable to break a test...

We could make this a non-blocking check, but we all know this will result in drive-by link fixes in unrelated PRs :-(. Checking new links in PRs is desirable though, and I suppose that wins.

Reply:
As one data point, we didn't actually have any stale links in our existing codebase, only ones that were invalid from the start.

Reply:
Another data point, the Operator SDK people check every link on every PR, because they say moving a file might break a link in an untouched file.

https://github.com/operator-framework/operator-sdk/pull/3048#discussion_r425848047

Reply:
Good points.  I prefer to perform the check on every PR to be safe. A PR could uncover an orthogonal issue but I think that should be uncommon and worth the inconvenience over possibly merging a PR with an issue. Then we'd have something broken publicly until a periodic check finds it and until someone actually notices and fixes it, which could be an arbitrary period of time.

Reply:
> Another data point, the Operator SDK people check every link on every PR, because they say moving a file might break a link in an untouched file.

Would this actually detect that? The check runs on unpublished documents... I’ll try a PR to find out.

Reply:
It does: https://github.com/submariner-io/submariner-website/pull/242